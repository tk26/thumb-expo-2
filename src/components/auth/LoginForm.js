import React, { Component } from 'react';
import { Text, Linking, Image } from 'react-native';
import { connect } from 'react-redux';
import { emailChanged, passwordChanged, loginUser, logoutUser } from '../../actions';
import { Card, CardSection, Input, Button, Spinner } from '../common';

class LoginForm extends Component {
  onEmailChange(text) {
    this.props.emailChanged(text);
  }

  onPasswordChange(text) {
    this.props.passwordChanged(text);
  }

  onButtonPress() {
    const { email, password } = this.props;

    this.props.loginUser({ email, password });
  }

  onLogoutPress() {
    this.props.logoutUser();
  }

  renderButton() {
    if (this.props.loading) {
      return <Spinner size="large" />;
    }

    return (
      <Button onPress={this.onButtonPress.bind(this)}>
        Login
      </Button>
    );
  }

  render() {
    return (
      <Card>
        <Image
          source={require('../../../assets/thumb-horizontal-logo.png')}
        />
        <CardSection>
            <Text>
                Log in to thumb
            </Text>
        </CardSection>        
        <CardSection>
          <Input
            label="Email"
            placeholder="email@university.edu"
            onChangeText={this.onEmailChange.bind(this)}
            value={this.props.email}
          />
        </CardSection>

        <CardSection>
          <Input
            secureTextEntry
            label="Password"
            placeholder="password"
            onChangeText={this.onPasswordChange.bind(this)}
            value={this.props.password}
          />
        </CardSection>

        <Text style={styles.errorTextStyle}>
          {this.props.error}
        </Text>

        <CardSection>
          {this.renderButton()}
        </CardSection>
        <CardSection>
          <Text style={{ color: 'blue' }} onPress={() => Linking.openURL('https://thumb-webapp.herokuapp.com/#/forgot')}>
              Forgot your password?
          </Text>
        </CardSection>
      </Card>
    );
  }
}

const styles = {
  errorTextStyle: {
    fontSize: 20,
    alignSelf: 'center',
    color: 'red'
  }
};

const mapStateToProps = ({ auth, profile }) => {
  const { email, password, error, loading, isLoggedIn } = auth;
  return { email, password, error, loading, isLoggedIn, profile };
};

export default connect(mapStateToProps, {
  emailChanged, passwordChanged, loginUser, logoutUser
})(LoginForm);