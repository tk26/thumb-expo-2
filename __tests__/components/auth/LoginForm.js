import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import LoginForm from '../../../src/components/auth/LoginForm';

const middlewares = [];
const mockStore = configureStore(middlewares);

const initialState = {
  auth: {
    email: '',
    password: '',
    token: '',
    isLoggedIn: false,
    error: '',
    loading: false
  },
  profile: {

  }
};

describe('Testing Login Form', () => {
  it('renders successfully with initial state', () => {
    const wrapper = shallow(
      <LoginForm />,
      { context: { store: mockStore(initialState) } },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
  it('renders successfully with error', () => {
    const auth = {...initialState.auth, error: 'test error message'};
    const state = {...initialState, auth};
    const wrapper = shallow(
      <LoginForm />,
      { context: { store: mockStore(state) } },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
  it('renders successfully when loading', () => {
    const auth = {...initialState.auth, loading: true};
    const state = {...initialState, auth};
    const wrapper = shallow(
      <LoginForm />,
      { context: { store: mockStore(state) } },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
