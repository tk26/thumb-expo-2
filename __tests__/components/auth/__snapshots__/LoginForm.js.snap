// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Login Form renders successfully when loading 1`] = `
<Container>
  <Header
    includeBackButton={true}
  />
  <Card>
    <Space
      height={30}
    />
    <CardSection>
      <Logo
        includeText={true}
        size="small"
      />
    </CardSection>
    <Space
      height={30}
    />
    <CardSection>
      <Input
        label="Email"
        onChangeText={[Function]}
        placeholder="email@university.edu"
        value=""
      />
    </CardSection>
    <PasswordInput
      onChangeText={[Function]}
      placeholder="password"
      value=""
    />
    <CardSection>
      <Spinner
        size="large"
      />
    </CardSection>
    <Space
      height={30}
    />
    <CardSection>
      <Link1
        linkText="Forgot Password?"
        onPress={[Function]}
      />
    </CardSection>
  </Card>
</Container>
`;

exports[`Testing Login Form renders successfully with error 1`] = `
<Container>
  <Header
    includeBackButton={true}
  />
  <Card>
    <Space
      height={30}
    />
    <CardSection>
      <Logo
        includeText={true}
        size="small"
      />
    </CardSection>
    <Space
      height={30}
    />
    <CardSection>
      <Input
        label="Email"
        onChangeText={[Function]}
        placeholder="email@university.edu"
        value=""
      />
    </CardSection>
    <PasswordInput
      onChangeText={[Function]}
      placeholder="password"
      value=""
    />
    <CardSection>
      <ErrorText>
        test error message
      </ErrorText>
    </CardSection>
    <CardSection>
      <Button
        onPress={[Function]}
      >
        log in
      </Button>
    </CardSection>
    <Space
      height={30}
    />
    <CardSection>
      <Link1
        linkText="Forgot Password?"
        onPress={[Function]}
      />
    </CardSection>
  </Card>
</Container>
`;

exports[`Testing Login Form renders successfully with initial state 1`] = `
<Container>
  <Header
    includeBackButton={true}
  />
  <Card>
    <Space
      height={30}
    />
    <CardSection>
      <Logo
        includeText={true}
        size="small"
      />
    </CardSection>
    <Space
      height={30}
    />
    <CardSection>
      <Input
        label="Email"
        onChangeText={[Function]}
        placeholder="email@university.edu"
        value=""
      />
    </CardSection>
    <PasswordInput
      onChangeText={[Function]}
      placeholder="password"
      value=""
    />
    <CardSection>
      <Button
        onPress={[Function]}
      >
        log in
      </Button>
    </CardSection>
    <Space
      height={30}
    />
    <CardSection>
      <Link1
        linkText="Forgot Password?"
        onPress={[Function]}
      />
    </CardSection>
  </Card>
</Container>
`;
